lib/pid_gatherer.rb

```diff
@@ -0,0 +1,114 @@
+module Argo
+  ##
+  # Gatherer of the PIDS, used for Argo indexing
+  class PidGatherer
+    def initialize(should_log_to_stdout=true)
+      if should_log_to_stdout
+        @logger = Logger.new(STDOUT)
+      else
+        @logger = Argo::Indexer.index_logger
+      end
+    end
+
+    ##
+    # @return [Array<Array<String>>]
+    def pid_lists_for_full_reindex
+      [uber_apo_pids, workflow_pids, agreement_pids, hydrus_uber_apo_pids, apo_pids, hydrus_apo_pids, collection_pids, hydrus_collection_pids, set_pids, remaining_pids]
+    end
+
+    ##
+    # @return [Array<String>]
+    def uber_apo_pids
+      @uber_apo_pids ||= ['druid:hv992ry2431']
+    end
+
+    ##
+    # @return [Array<String>]
+    def workflow_pids
+      @workflow_pids ||= pids_for_model_type '<info:fedora/afmodel:Dor_WorkflowObject>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def agreement_pids
+      @agreement_pids ||= pids_for_model_type '<info:fedora/afmodel:agreement>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def hydrus_uber_apo_pids
+      @hydrus_uber_apo_pids || ['druid:zw306xn5593']
+    end
+
+    ##
+    # @return [Array<String>]
+    def apo_pids
+      @apo_pids ||= pids_for_model_type '<info:fedora/afmodel:Dor_AdminPolicyObject>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def hydrus_apo_pids
+      @hydrus_apo_pids ||= pids_for_model_type '<info:fedora/afmodel:Hydrus_AdminPolicyObject>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def collection_pids
+      @collection_pids ||= pids_for_model_type '<info:fedora/afmodel:Dor_Collection>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def hydrus_collection_pids
+      @hydrus_collection_pids ||= pids_for_model_type '<info:fedora/afmodel:Hydrus_Collection>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def set_pids
+      @set_pids ||= pids_for_model_type '<info:fedora/afmodel:Dor_Set>'
+    end
+
+    ##
+    # @return [Array<String>]
+    def all_pids
+      @logger.info "querying fedora for all pids..."
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.